scala/private/rules/scala_export.bzl

```diff
@@ -0,0 +1,132 @@
+    the template file:
+
+      * `{groupId}`: Replaced with the maven coordinates group ID.
+      * `{artifactId}`: Replaced with the maven coordinates artifact ID.
+      * `{version}`: Replaced by the maven coordinates version.
+      * `{type}`: Replaced by the maven coordintes type, if present (defaults to "jar")
+      * `{dependencies}`: Replaced by a list of maven dependencies directly relied upon
+        by scala_library targets within the artifact.
+
+    The "edges" of the artifact are found by scanning targets that contribute to
+    runtime dependencies for the following tags:
+
+      * `maven_coordinates=group:artifact:type:version`: Specifies a dependency of
+        this artifact.
+      * `maven:compile-only`: Specifies that this dependency should not be listed
+        as a dependency of the artifact being generated.
+
+    To skip generation of the javadoc jar, add the `no-javadocs` tag to the target.
+    To skip generation of the scaladoc jar, add the `no-scaladocs` tag to the target.
+
+    Generated rules:
+      * `name`: A `scala_library` that other rules can depend upon.
+      * `name-docs`: A javadoc jar file.
+      * `name-scaladocs`: A scaladoc jar file.
+      * `name-pom`: The pom.xml file.
+      * `name.publish`: To be executed by `bazel run` to publish to a maven repo.
+
+    Args:
+      name: A unique name for this target
+      maven_coordinates: The maven coordinates for this target.
+      pom_template: The template to be used for the pom.xml file.
+      deploy_env: A list of labels of java targets to exclude from the generated jar
+      visibility: The visibility of the target
+      publish_maven_metadata: Whether to publish a maven-metadata.xml
+      kwargs: These are passed to [`scala_library`](https://github.com/bazelbuild/rules_scala/blob/master/docs/scala_library.md),
+        and so may contain any valid parameter for that rule.
+    """
+    maven_coordinates_tags = ["maven_coordinates=%s" % maven_coordinates]
+    lib_name = "%s-lib" % name
+
+    javadocopts = kwargs.pop("javadocopts", None)
+    doc_resources = kwargs.pop("doc_resources", [])
+    classifier_artifacts = kwargs.pop("classifier_artifacts", {})
+
+    updated_deploy_env = [] + deploy_env
+    for lib in SCALA_LIBS:
+        if lib not in deploy_env:
+            updated_deploy_env.append(lib)
+
+    scala_library(
+        name = lib_name,
+        tags = tags + maven_coordinates_tags,
+        testonly = testonly,
+        **kwargs
+    )
+
+    if "no-scaladocs" not in tags:
+        scaladocs_name = name + "-scaladocs-html"
+        scala_doc(
+            name = scaladocs_name,
+            deps = [":" + lib_name],
+        )
+
+        scaladocs_jar_name = name + "-scaladocs"
+        create_jar(
+            name = scaladocs_jar_name,
+            inputs = [":" + scaladocs_name] + doc_resources,
+            out = name + "-scaladocs.jar",
+        )
+        classifier_artifacts["scaladoc"] = scaladocs_jar_name
```

Comment:
Shouldn't that be a `javadoc`? `scaladoc` is definitely a thing, but `javadoc` is standard classifier for Maven artifacts containing documentation.

As an example all artifacts published to Maven Sonatype need to have a required `javadoc` artifact, nevertheless if that's a Scala or Java library. 

It also seems like maven_export would try to build `javadoc`, I guess it might fail for Scal sources, so maybe we should prevent it from doing so. 
https://github.com/bazel-contrib/rules_jvm_external/blob/a23c85cf0af3a2b10d3d84da7a0160261b38e2bb/private/rules/java_export.bzl#L307-L328

I expect that as result we should have the same outputs, as can be found by sbt (or other Scala dedicated build tools) releases: `–javadoc`, `-sources`, actual jar and soruces